src/main/java/gregtech/common/metatileentities/multi/steam/MetaTileEntitySteamBlastFurnace.java

```diff
@@ -0,0 +1,73 @@
+package gregtech.common.metatileentities.multi.steam;
+
+import gregtech.api.capability.impl.SteamMultiWorkable;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.metatileentity.MetaTileEntityHolder;
+import gregtech.api.metatileentity.multiblock.IMultiblockPart;
+import gregtech.api.metatileentity.multiblock.MultiblockAbility;
+import gregtech.api.metatileentity.multiblock.RecipeMapSteamMultiblockController;
+import gregtech.api.multiblock.BlockPattern;
+import gregtech.api.multiblock.FactoryBlockPattern;
+import gregtech.api.recipes.RecipeMaps;
+import gregtech.api.render.ICubeRenderer;
+import gregtech.api.render.OrientedOverlayRenderer;
+import gregtech.api.render.Textures;
+import gregtech.common.ConfigHolder;
+import gregtech.common.blocks.BlockMetalCasing;
+import gregtech.common.blocks.MetaBlocks;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.ResourceLocation;
+
+import javax.annotation.Nonnull;
+
+import static gregtech.api.render.Textures.BRONZE_PLATED_BRICKS;
+import static gregtech.api.render.Textures.SOLID_STEEL_CASING;
+
+public class MetaTileEntitySteamBlastFurnace extends RecipeMapSteamMultiblockController {
+
+    private static final MultiblockAbility<?>[] ALLOWED_ABILITIES = {
+            MultiblockAbility.STEAM_IMPORT_ITEMS, MultiblockAbility.STEAM_EXPORT_ITEMS, MultiblockAbility.STEAM
+    };
+
+    public MetaTileEntitySteamBlastFurnace(ResourceLocation metaTileEntityId) {
+        super(metaTileEntityId, RecipeMaps.STEAM_BLAST_FURNACE_RECIPES, CONVERSION_RATE);
+        this.recipeMapWorkable = new SteamMultiWorkable(this, CONVERSION_RATE, 4);
+    }
+
+    @Override
+    public MetaTileEntity createMetaTileEntity(MetaTileEntityHolder metaTileEntityHolder) {
+        return new MetaTileEntitySteamBlastFurnace(metaTileEntityId);
+    }
+
+    @Override
+    protected BlockPattern createStructurePattern() {
+        return FactoryBlockPattern.start()
+                .aisle("XXX", "XXX", "XXX", "XXX")
+                .aisle("XXX", "X#X", "X#X", "X#X")
+                .aisle("XXX", "XYX", "XXX", "XXX")
+                .where('Y', selfPredicate())
+                .where('X', statePredicate(getCasingState()).or(abilityPartPredicate(ALLOWED_ABILITIES)))
+                .where('#', isAirPredicate())
+                .build();
+    }
+
+    public IBlockState getCasingState() {
+        return ConfigHolder.U.steelSteamMultiblocks ? MetaBlocks.METAL_CASING.getState(BlockMetalCasing.MetalCasingType.STEEL_SOLID) : MetaBlocks.METAL_CASING.getState(BlockMetalCasing.MetalCasingType.BRONZE_BRICKS);
+    }
+
+    @Override
+    public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart) {
+        return ConfigHolder.U.steelSteamMultiblocks ? SOLID_STEEL_CASING : BRONZE_PLATED_BRICKS;
+    }
+
+    @Nonnull
+    @Override
+    protected OrientedOverlayRenderer getFrontOverlay() {
+        return Textures.PRIMITIVE_BLAST_FURNACE_OVERLAY;
```

Comment:
We should probably make a new FrontOverlay for this, incase texturepack makers want to modify things. You can simply reuse the current primitive blast furnace textures though, to have that texture there incase someone makes a new one.